lib/rails_state_machine/state_machine.rb

```diff
@@ -2,9 +2,12 @@ module RailsStateMachine
   class StateMachine
     attr_reader :model
 
-    def initialize(model, state_attribute)
+    StateConstantAlreadyDefinedError = Class.new(StandardError)
```

Comment:
Internally, we clash because of the constant definition, but the error is actually that the state name is already in use on the current model. The error class' name does not need to contain internals.

```suggestion
    StateAlreadyDefinedError = Class.new(StandardError)
```